### PR TITLE
[MM-27019] Migrate PR handler and add some tests

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
This PR migrates `pullRequestEventHandler` and add some testing around data validation but specific tests for events are not included.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-27019

### CODE REVIEW

File: server/pull_request_handler.go [hunk 1]
```diff
@@ -157,29 +181,45 @@ func (s *Server) handlePullRequestEvent(ctx context.Context, event *PullRequestE
 		}
 	}
 
-	s.checkPullRequestForChanges(ctx, pr)
+	changed, err := s.checkPullRequestForChanges(ctx, pr)
+	if err != nil {
+		mlog.Error("Could not check changes for PR", mlog.Err(err))
```
Comment (ethervoid) [inline server/pull_request_handler.go:186]: Maybe we should return 400 or 500 error here

File: server/pull_request_handler.go [hunk 2]
```diff
@@ -157,29 +181,45 @@ func (s *Server) handlePullRequestEvent(ctx context.Context, event *PullRequestE
...
 		}
 
-		return
+		return false, nil
 	}
 
-	prHasChanges := false
+	var prHasChanges bool
 
 	for _, label := range pr.Labels {
```
Comment (ethervoid) [inline server/pull_request_handler.go:224]: Maybe we can extract this part of the code and create a new function that receives two string slices and returns the diff between them so we can reduce the amount of boilerplate here. What do you think?

**Comment by agnivade** [comment]
Sorry this will get a bit delayed on review.

File: server/pull_request_handler.go (hunk 1 above)
Comment (isacikgoz) [reply server/pull_request_handler.go:186]: Makes sense, I also thought to return `500` for this but didn't want to introduce behavioral change. But I'm open to change it 👍

Comment (ethervoid) [reply server/pull_request_handler.go:186]: I'll do it. All the HTTP errors are new but I think is necessary to be aware of problems. Those errors would be reflected in the metrics so it's a good thing

File: server/pull_request_handler.go [hunk 3]
```diff
@@ -248,13 +292,15 @@ func (s *Server) checkPullRequestForChanges(ctx context.Context, pr *model.PullR
 	if prHasChanges {
 		mlog.Info("pr has changes", mlog.Int("pr", pr.Number))
```
Comment (agnivade) [inline server/pull_request_handler.go:293]: We should remove this. The logging is already happening at the upper level.

File: server/pull_request_handler.go (hunk 2 above)
Comment (isacikgoz) [reply server/pull_request_handler.go]: Yeah, makes sense to avoid repetition.

**Pull request merged**